webapp/javascript/redux/reducers/settings.ts

```diff
@@ -34,15 +35,29 @@ type ApiKeysState = {
   data?: APIKeys;
 };
 const apiKeysInitialState: ApiKeysState = { type: 'pristine', data: undefined };
+let reloadApiKeysAbortController: AbortController | undefined;
 
 export const reloadApiKeys = createAsyncThunk(
   'newRoot/reloadAPIKeys',
   async (_, thunkAPI) => {
-    const res = await fetchAPIKeys();
+    if (reloadApiKeysAbortController) {
+      reloadApiKeysAbortController.abort();
+    }
+    reloadApiKeysAbortController = new AbortController();
+
+    thunkAPI.signal = reloadApiKeysAbortController.signal;
+
+    const res = await fetchAPIKeys(reloadApiKeysAbortController);
+
     if (res.isOk) {
       return Promise.resolve(res.value);
     }
 
+    if (res.isErr && res.error instanceof RequestIncompleteError) {
+      return Promise.reject(res);
+    }
+
+    console.log(res);
```

Comment:
One problem with this is that the return type changed to be either `Promise<typeof res.error | typeof res | typeof res.value>`. From my understanding we can still `Promise.reject(res.error)` here right?

Another thing is I think we can make this more clear this by checking for `e.name == 'AbortError'` here https://github.com/pyroscope-io/pyroscope/blob/main/webapp/javascript/services/base.ts#L97, then we know exactly it's an `AbortError` (request has ben aborted) and not an `RequestIncompleteError` (eg user is without network)

Reply:
yep, noted